src/raster/rasterband.rs

```diff
@@ -612,3 +621,144 @@ impl ColorInterpretation {
         _string(rv)
     }
 }
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum PaletteInterpretation {
+    Gray,
+    Rgba,
+    Cmyk,
+    Hls,
+}
+
+impl PaletteInterpretation {
+    fn from_c_int(palette_interpretation: GDALPaletteInterp::Type) -> Self {
+        match palette_interpretation {
+            GDALPaletteInterp::GPI_Gray => Self::Gray,
+            GDALPaletteInterp::GPI_RGB => Self::Rgba,
+            GDALPaletteInterp::GPI_CMYK => Self::Cmyk,
+            GDALPaletteInterp::GPI_HLS => Self::Hls,
+            _ => unreachable!("GDAL has implemented a new type of `GDALPaletteInterp`"),
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub struct GrayEntry {
+    pub g: i16,
```

Comment:
Similar to the `i32` / `usize` discussion: do you know how often these happen to be negative? I mean, if I have a 16-bit RGB image, I'll still want positive numbers here.

Anyway, this isn't blocking, I was just wondering how GDAL expects these to be used.

Reply:
I don't think they would ever be negative but I thought it might be best to leave them the same as what gdal_sys provides. I suppose elevation data could be negative but 
I can't imagine that ever being palletized.